searxng/searxng
# [mod] add make targets to manage the build files in the /searx/static directory

## What does this PR do?

The idea is to avoid conflict when there are differents branches with changes are made on the static files.
A solution is to ask the administrators to build the files from the sources, but it requires to install the npm dependencies.
So the solution in this commit keep the sources and the build files in the same git repository.

In one branch, the modification of the source (*.less, *.js) are commited without the built files.
The built files are commited in a uniq commit, with a commit message "Static build"
In case of merge or rebase, this commit can be dropped.

New make targets:
* `static.build.commit.drop`:
    * drop the last "Static build".
    * The command checks that there are only build files in the commit.
* `static.build.commit` :
    * call "make static.build.commit.drop"
    * call "make themes.all"
    * commit the files
* `static.git.restore.staged` :
    * `git restore --staged <build files>`
* `static.git.restore` :
    * `git restore <build files>`

#### Use cases

* Typical
  * `git add .` # the build files are included
  * `make static.git.restore` # restore them.
  * `git commit`  # commit only the source
  * `make static.build.commit`  # build & commit the built files from the sources

* Rebase with change of the static files in the current branch and in the master branch:
  * `make static.build.commit.drop`
  * `git pull --rebase origin/master`
  * `make static.build.commit`

## Why is this change important?





## How to test this PR locally?

* change some *.less files
* call `make themes`
* call `static.git.restore`
* check only the build files has been restored.
* call `make static.build.commit`
* check the build commit  is here
* call `make static.build.commit.drop`
* check the build commit has been removed
* add a dummy commit
* call `make static.build.commit.drop`
* check the last has not been dropped

## Author's checklist



## Related issues

* https://github.com/searxng/searxng/issues/137
* The last step for https://github.com/searxng/searxng/wiki/Milestones#milestone-19---data-upgrade--build-process
